src/software/ai/hl/stp/tactic/shadow_enemy/shadow_enemy_fsm.h

```diff
@@ -0,0 +1,215 @@
+
+        const auto update_e = event<Update>;
+
+        /**
+         * Guard that checks if the enemy threat has ball
+         *
+         * @param event ShadowEnemyFSM::Update
+         *
+         * @return if the ball has been have_possession
+         */
+        const auto enemy_threat_has_ball = [](auto event) {
+            std::optional<EnemyThreat> enemy_threat_opt =
+                event.control_params.enemy_threat;
+            if (enemy_threat_opt.has_value())
+            {
+                return enemy_threat_opt.value().has_ball;
+            };
+            LOG(WARNING) << "Enemy threat not initialized for robot "
+                         << event.common.robot.id() << "\n";
+            return false;
+        };
+
+        /**
+         * Action to block the pass to our shadowee
+         *
+         *
+         * @param event ShadowEnemyFSM::Update
+         */
+        const auto block_pass = [](auto event) {
+            std::optional<EnemyThreat> enemy_threat_opt =
+                event.control_params.enemy_threat;
+            auto ball_position = event.common.world.ball().position();
+            auto face_ball_orientation =
+                (ball_position - event.common.robot.position()).orientation();
+            Point position_to_block =
+                ball_position +
+                (event.common.world.field().friendlyGoalCenter() - ball_position)
+                    .normalize(event.control_params.shadow_distance);
+            if (enemy_threat_opt.has_value())
+            {
+                position_to_block =
+                    findBlockPassPoint(ball_position, enemy_threat_opt.value().robot,
+                                       event.control_params.shadow_distance);
+            };
+
+            event.common.set_intent(std::make_unique<MoveIntent>(
+                event.common.robot.id(), position_to_block, face_ball_orientation, 0,
+                DribblerMode::OFF, BallCollisionType::AVOID,
+                AutoChipOrKick{AutoChipOrKickMode::OFF, 0},
+                MaxAllowedSpeedMode::PHYSICAL_LIMIT, 0.0));
+        };
+
+        /**
+         * Action to block the shot from our shadowee
+         *
+         *
+         * @param event ShadowEnemyFSM::Update
+         */
+        const auto block_shot = [this](auto event,
+                                       back::process<MoveFSM::Update> processEvent) {
+            std::optional<EnemyThreat> enemy_threat_opt =
+                event.control_params.enemy_threat;
+            auto ball_position = event.common.world.ball().position();
+            auto face_ball_orientation =
+                (ball_position - event.common.robot.position()).orientation();
+            Point position_to_block =
+                ball_position +
+                (event.common.world.field().friendlyGoalCenter() - ball_position)
+                    .normalize(event.control_params.shadow_distance);
```

Comment:
Can we add comments so we're explicit that this is a default blocking position if we don't have an enemy_threat?